flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkSqlParserImplTest.java

```diff
@@ -611,6 +611,70 @@ void testAlterTableAddWatermark() {
                 .fails("Multiple WATERMARK statements is not supported yet.");
     }
 
+    @Test
+    void testAlterTableAddDistribution() {
+        sql("alter table t1 add (DISTRIBUTION BY HASH(a, h) INTO 6 BUCKETS\n)")
```

Comment:
Parenthesis should not be allowed. Take a look at the grammar in the Flink docs:
```
ADD { <schema_component> | (<schema_component> [, ...]) | [IF NOT EXISTS] <partition_component> [<partition_component> ...]}
```

Distribution is not a schema component. It deserves its own branch.  Do we support parenthesis for the partition component?

Reply:
I've sorted out the parser changes for this.